clang/test/SemaSYCL/remove-restriction-builtin-assume-aligned.cpp

```diff
@@ -0,0 +1,21 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -verify -pedantic -fsyntax-only %s
+// This test checks if __builtin_assume_aligned does not throw an error when
+// called from within device code.
+
+#include "sycl.hpp"
+
+using namespace sycl;
+queue q;
+
+int main() {
+  int *Ptr[2];
+  // expected-no-diagnostics
+  q.submit([&](handler &h) {
+    h.single_task<class kernelA>([=]() {
+    int *APtr = (int *)__builtin_assume_aligned(Ptr, 32);
```

Comment:
Is the variadic functions handled as expected in device code? I do not remember why we prohibited this in the first place.

Reply:
Not variadic functions in general.  This builtin though is just an informational to the BE, so should be fine.   I think.